src/openmc_driver.cpp

```diff
@@ -113,69 +127,72 @@ std::vector<CellHandle> OpenmcDriver::find(const std::vector<Position>& position
   std::vector<CellHandle> handles;
   handles.reserve(positions.size());
 
-  std::unordered_map<CellInstance, CellHandle> cell_index;
-
   for (const auto& r : positions) {
     // Determine cell instance corresponding to global element
     CellInstance c{r};
 
     // If this cell instance hasn't been saved yet, add it to cells_ and
     // keep track of what index it corresponds to
-    if (cell_index.find(c) == cell_index.end()) {
-      cell_index[c] = cells_.size();
-      cells_.push_back(c);
-    }
-    auto h = cell_index.at(c);
```

Comment:
These changes are relevant to your discussion about `cells_` in OpenmcDriver and ShiftDriver.  If I wanted to keep these data members more-or-less intact, I might be able to do something like:
```
if (cell_index.find(c) == cell_index.end()) {
       cell_index[c] = c.get_handle();
       cells_.push_back(c);
     }
     auto h = cell_index.at(c);
```
It would require me to change my usage of `cells_` in the rest of the PR.  But if you think it would clarify the intent, I'd be happy to do it.